### PR TITLE
Fixing grasp inside and eight tasks

### PR DESCRIPTION
**Pull request by wmcclinton**
Added GraspInside NSRT and made minor changes to get eight tasks working: `opening_presents`, `locking_every_window`, `re-shelving_library_book`, `sorting_books`, `clearing_the_table_after_dinner`, `opening_packages`, `locking_every_door`, and some scenes for `organizing_file_cabinet`.

### CODE REVIEW

File: predicators/envs/behavior.py [hunk 1]
```diff
@@ -235,8 +235,8 @@ def _get_task_goal(self) -> Set[GroundAtom]:
         # Currently assumes that the goal is a single AND of
         # ground atoms (this is also assumed by the planner).
         goal = set()
-        assert len(
-            self.igibson_behavior_env.task.ground_goal_state_options) == 1
+        # assert len(
+        #     self.igibson_behavior_env.task.ground_goal_state_options) == 1
```
Comment (NishanthJKumar) [inline predicators/envs/behavior.py]: why did you comment out this assertion?

File: predicators/ground_truth_nsrts.py [hunk 2]
```diff
@@ -3012,15 +3016,32 @@ def place_inside_obj_pos_sampler(
...
                 ontop = _get_lifted_atom("ontop", [target_obj, surf_obj])
-                preconditions = {handempty, targ_reachable, ontop}
+                inside = _get_lifted_atom("inside", [target_obj, surf_obj])
+                preconditions_ontop = {handempty, targ_reachable, ontop}
+                preconditions_inside = {handempty, targ_reachable, inside}
                 add_effects = {targ_holding}
-                delete_effects = {handempty, ontop, targ_reachable}
+                delete_effects_ontop = {handempty, ontop, targ_reachable}
+                delete_effects_inside = {handempty, inside, targ_reachable}
+                # NSRT for grasping an object from ontop an object
```
Comment (NishanthJKumar) [inline predicators/ground_truth_nsrts.py]: NIT: add a period at tend of this comment.

File: predicators/ground_truth_nsrts.py [hunk 3]
```diff
@@ -3012,15 +3016,32 @@ def place_inside_obj_pos_sampler(
...
                     add_effects,
-                    delete_effects,
+                    delete_effects_ontop,
+                    set(),
+                    option,
+                    option_vars,
+                    grasp_obj_param_sampler,
+                )
+                nsrts.add(nsrt)
+                # NSRT for grasping an object from inside an object
```
Comment (NishanthJKumar) [inline predicators/ground_truth_nsrts.py]: NIT: add a period at tend of this comment.

File: predicators/ground_truth_nsrts.py [hunk 4]
```diff
@@ -2959,6 +2959,10 @@ def place_inside_obj_pos_sampler(
         split_name = option.name.split("-")
         base_option_name = split_name[0]
         option_arg_type_names = split_name[1:]
+        # Edge case t-shirt gets parsed wrong with split("")
```
Comment (NishanthJKumar) [inline predicators/ground_truth_nsrts.py]: NIT: add a period at tend of this comment.

File: predicators/envs/behavior.py [hunk 5]
```diff
@@ -251,6 +251,9 @@ def _get_task_goal(self) -> Set[GroundAtom]:
             else:
                 bddl_name = head_expr.terms[0]  # untyped
                 obj_start_idx = 1
+            # For onfloor we will just use ontop of floor
```
Comment (NishanthJKumar) [inline predicators/envs/behavior.py]: NIT: add a period at tend of this comment.

File: predicators/envs/behavior.py [hunk 6]
```diff
@@ -251,6 +251,9 @@ def _get_task_goal(self) -> Set[GroundAtom]:
...
+            if bddl_name == 'onfloor':
+                bddl_name = 'ontop'
```
Comment (NishanthJKumar) [inline predicators/envs/behavior.py:259]: do we care? Why is this predicate important to get from the initial state?

**Review by NishanthJKumar — COMMENTED** [review]
Looks mostly good; should be gtg after addressing a few comments!

File: predicators/envs/behavior.py (hunk 6 above)
Comment (wmcclinton) [reply predicators/envs/behavior.py:259]: It is in the goal too.

**Review by NishanthJKumar — APPROVED** [review]
LGTM!

**Pull request merged**